pkg/controller/common/metadata.go

```diff
@@ -27,6 +27,15 @@ const (
 
 	// InternalKey is used to identify the resource as being internal to the mesh itself (i.e. should not be applied to members)
 	InternalKey = MetadataNamespace + "/internal"
+
+	// KubernetesAppNamespace is the common namespace for application information
+	KubernetesAppNamespace = "app.kubernetes.io"
+	KubernetesAppNameKey = KubernetesAppNamespace + "/name"
+	KubernetesAppInstanceKey = KubernetesAppNamespace + "/instance"
+	KubernetesAppVersionKey = KubernetesAppNamespace + "/version"
+	KubernetesAppComponentKey = KubernetesAppNamespace + "/component"
+	KubernetesAppPartOfKey = KubernetesAppNamespace + "/part-of"
+	KubernetesAppManagedByKey = KubernetesAppNamespace + "/managed-by"
```

Comment:
Nitpick: I prefer to define externally visible constants as a single string, so that any users searching for the constant in the source can easily find it.

Reply:
For me, this saves typo's.  I think I've seen it both ways.  I usually end up searching for the root of the key, then search using the variable name(s).